gossip/emitter/throttler/throttler_test.go

```diff
@@ -0,0 +1,241 @@
+
+			tests[fmt.Sprintf(
+				"lastSeenAt=%d NonDominatingTimeout=%d",
+				lastSeenAt,
+				NonDominatingTimeout,
+			)] = testCase{
+				NonDominatingTimeout: NonDominatingTimeout,
+				lastAttendance: validatorAttendance{
+					lastSeenSeq: 123,
+					lastSeenAt:  lastSeenAt,
+					online:      true,
+				},
+				expectedOnline: lastSeenAt+NonDominatingTimeout > currentAttempt,
+			}
+		}
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			ctrl := gomock.NewController(t)
+			validators := makeValidatorsFromStakes(100)
+			world := NewMockWorldReader(ctrl)
+			world.EXPECT().GetEpochValidators().Return(validators, idx.Epoch(0))
+			world.EXPECT().GetLastEvent(idx.ValidatorID(1)).Return(makeEventWithSeq(123))
+
+			config := config.ThrottlerConfig{
+				Enabled:                true,
+				DominantStakeThreshold: 0.75, // this test is stake agnostic
+				DominatingTimeout:      3,    // fix short timeout
+				NonDominatingTimeout:   test.NonDominatingTimeout,
+			}
+
+			attendanceList := newAttendanceList()
+			attendanceList.attendance[1] = test.lastAttendance
+
+			// notice empty lastDominantSet - validator is suppressed
+			attendanceList.updateAttendance(world, config, nil, currentAttempt)
+
+			require.Equal(t, test.expectedOnline, attendanceList.isOnline(1))
+		})
+	}
+}
+
+func TestThrottler_updateAttendance_OfflineValidatorsComeBackOnlineWithAnyNewSeqNumber(t *testing.T) {
+	t.Parallel()
+
+	const currentAttempt = 100
+	const lastSeenSeq = 122
+
+	type testCase struct {
+		NonDominatingTimeout config.Attempt
+		lastAttendance       validatorAttendance
+	}
+	tests := make(map[string]testCase)
+	for lastSeenAt := config.Attempt(1); lastSeenAt <= currentAttempt; lastSeenAt++ {
+		for _, DominatingTimeout := range []config.Attempt{1, 2, 3, 8, 10} {
+			for _, NonDominatingTimeout := range []config.Attempt{1, 2, 3, 8, 10} {
+
+				tests[fmt.Sprintf(
+					"lastSeenAt=%d NonDominatingTimeout=%d DominatingTimeout=%d",
+					lastSeenAt,
+					NonDominatingTimeout,
+					DominatingTimeout,
+				)] = testCase{
+					NonDominatingTimeout: DominatingTimeout,
+					lastAttendance: validatorAttendance{
+						lastSeenSeq: lastSeenSeq,
+						lastSeenAt:  lastSeenAt,
+					},
+				}
+			}
+		}
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			ctrl := gomock.NewController(t)
+			validators := makeValidatorsFromStakes(100)
+			world := NewMockWorldReader(ctrl)
+			world.EXPECT().GetEpochValidators().Return(validators, idx.Epoch(0))
+			world.EXPECT().GetLastEvent(idx.ValidatorID(1)).
+				Return(makeEventWithSeq(lastSeenSeq + 1))
+
+			config := config.ThrottlerConfig{
+				Enabled:                true,
+				DominantStakeThreshold: 0.75,
+				DominatingTimeout:      3,
+				NonDominatingTimeout:   test.NonDominatingTimeout,
+			}
+
+			attendanceList := newAttendanceList()
+			attendanceList.attendance[1] = test.lastAttendance
+
```

Comment:
nit: add a 
```Go
require.False(t, attendenceList.isOnline(1))
```
here to signal that all this setup should make sure that validator 1 is not online before the update.